src/JsonSchema/Validator.php

```diff
@@ -45,10 +45,10 @@ public function check($value, $schema = null, JsonPointer $path = null, $i = nul
      *
      * {@inheritDoc}
      */
-    public function coerce(&$value, $schema = null, JsonPointer $path = null, $i = null)
+    public function coerce(&$value, $schema = null, JsonPointer $path = null, $i = null, $default = false)
```

Comment:
Rather than trail another argument on the end of `check` and `coerce`, you might consider adding a config option to drive this. Something like:

```PHP
abstract class Constraint implements ConstraintInterface
{
//...
    const CHECK_MODE_NORMAL = 0x00000001;
    const CHECK_MODE_TYPE_CAST = 0x00000002;
    const CHECK_MODE_USE_DEFAULTS = 0x00000004;
```

That way you would only really have to change the code in one place and not have to chase down all the different paths like you're currently doing (and wonder if you missed any).